src/Form.php

```diff
@@ -434,25 +448,55 @@ public function ajaxSubmit()
             ->setStyle(['display'=>'none']);
 
         $cb->set(function () {
+            $caught = function ($e) {
+                return new jsExpression('$([html]).modal("show")', [
+                    'html'=> '<div class="ui fullscreen modal"> <i class="close icon"></i> <div class="header"> '.
+                    htmlspecialchars(get_class($e)).
+                    ' </div> <div class="content"> '.
+                    ($e instanceof \atk4\core\Exception ? $e->getHTML() : nl2br(htmlspecialchars($e->getMessage())))
+                    .' </div> </div>',
+                ]);
+            };
+
             try {
-                $this->loadPOST();
-                $response = $this->hook('submit');
-                if (!$response) {
-                    if (!$this->model instanceof \atk4\ui\misc\ProxyModel) {
-                        $this->model->save();
-
-                        return $this->success('Form data has been saved');
-                    } else {
-                        return new jsExpression('console.log([])', ['Form submission is not handled']);
+                try {
```

Comment:
Do we really need 3 inherited try blocks?
We can use
```
try {
    ...
} catch (\atk4\data\ValidationException $val) {
    ...
} catch (\Error $e) {
    ...
} catch (\Exception $e) {
    ...
}
```

Reply:
I leave this to you to refactor later, can you do it? I am not sure if php 5.6 supports multiple catch blocks, but 7.1 certainly good by adding https://wiki.php.net/rfc/multiple-catch.

Reply:
I didn't mean multiple-catch like catch (\Error $1 || \Exception $e2) syntax. I mean not nesting try/catch blocks.